components/local-app/pkg/bastion/bastion.go

```diff
@@ -142,7 +142,10 @@ func (b *Bastion) fullUpdate(uchan chan<- *gitpod.WorkspaceInstance) {
 func (b *Bastion) acceptUpdates(updates chan *gitpod.WorkspaceInstance) {
 	for u := range updates {
 		ws, ok := b.workspaces[u.ID]
-		if !ok && u.Status.Phase != "stopping" {
+		if !ok {
+			if u.Status.Phase == "stopping" || u.Status.Phase == "stopped" {
```

Comment:
Interesting. I never saw a stopped phase even though I expected it (but then I didn't look too hard). I saw the ssh session closed and the connection to supervisor lost before it stopped. Agree it should be handled.